tests/RedisMutexTest.php

```diff
@@ -2,168 +2,129 @@
-        ]);
-        $mutex2 = $this->getMutexInstance($connection);
+        sleep(3);
 
-        $mutex1->acquire($testLockName);
-        $this->assertTrue($mutex2->acquire($testLockName, 300));
+        $this->assertMutexKeyInRedisExistsOrNot(false);
+        $this->assertFalse($mutex->release(static::$mutexName));
+        $this->assertMutexKeyInRedisExistsOrNot(false);
     }
 
-    public function testLockTimeoutFailure()
-    {
-        $testLockName = 'the_lock';
-
-        $connection = $this->getConnection(true);
-
-        $mutex1 = $this->getMutexInstance($connection, [
-            'lockExpire' => 10000,
-        ]);
-        $mutex2 = $this->getMutexInstance($connection);
-
-        $mutex1->acquire($testLockName);
-        $this->assertFalse($mutex2->acquire($testLockName, 200));
+    protected function setUp() {
+        parent::setUp();
+        $databases = TestCase::getParam('databases');
+        $params = isset($databases['redis']) ? $databases['redis'] : null;
+        if ($params === null) {
+            $this->markTestSkipped('No redis server connection configured.');
+            return;
+        }
+
+        $connection = new Connection($params);
+        $this->mockApplication(['components' => ['redis' => $connection]]);
     }
 
-    public function testLockExpire()
+    /**
+     * @return Mutex
+     * @throws \yii\base\InvalidConfigException
+     */
+    protected function createMutex()
     {
-        $testLockName = 'the_lock';
-        $lockKeyPrefix = 'foo_';
-
-        $connection = $this->getConnection(true);
-
-        $mutex = $this->getMutexInstance($connection, [
-            'lockExpire' => 100,
-            'keyPrefix' => $lockKeyPrefix,
+        return Yii::createObject([
+            'class' => Mutex::className(),
+            'expire' => 2.5,
+            'keyPrefix' => static::$mutexPrefix
         ]);
-
-        $mutex->acquire($testLockName);
-        usleep(300000);
-        $this->assertFalse($mutex->release($testLockName));
-        $this->assertNull($connection->get($lockKeyPrefix . $testLockName));
     }
 
-    public function testAcquireExistingLock()
+    protected function getKey($name)
     {
-        $lockKeyPrefix = 'foo_';
-        $testLockName = 'the_lock';
-
-        $connection = $this->getConnection(true);
-        $mutex = $this->getMutexInstance($connection, [
-            'keyPrefix' => $lockKeyPrefix,
-        ]);
-
-        $mutex->acquire($testLockName);
-
-        $this->assertFalse($mutex->acquire($testLockName));
+        if (!isset(self::$_keys[$name])) {
+            $mutex = $this->createMutex();
+            $method = new \ReflectionMethod($mutex, 'calculateKey');
+            $method->setAccessible(true);
+            self::$_keys[$name] = $method->invoke($mutex, $name);
+        }
+        return self::$_keys[$name];
     }
 
-    public function testReleaseNonExistingLock()
+    protected function assertMutexKeyInRedisExistsOrNot($exists)
```

Comment:
this is ackward? the name of this assertion implies true at all times ;). Please split in 2 assertion methods without arguments